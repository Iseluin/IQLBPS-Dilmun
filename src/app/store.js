
import {
  configureStore
} from '@reduxjs/toolkit';
import userReducer from '../features/user/userSlice';
import itemReducer from '../features/slices/item'

export const store = configureStore({
  reducer: {
    user: userReducer,
    item: itemReducer
  },
});
