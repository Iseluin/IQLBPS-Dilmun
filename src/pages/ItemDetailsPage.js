import React from 'react'
import CustomButton from '../components/CustomButton'

const ItemDetailsPage = () => {
    return (
        <div className='flex text-background mt-11'>
            <div>
                <img src="https://images.unsplash.com/photo-1505740420928-5e560c06d30e?ixlib=rb-4.0.3&ixid=MnwxMjA3fDB8MHxzZWFyY2h8M3x8cHJvZHVjdHxlbnwwfHwwfHw%3D&w=1000&q=80" alt="headphones" />
            </div>
            <div className=' max-w-md ml-auto mr-auto flex flex-col gap-12 rounded pt-5 px-5 bg-primary '>
                <div>
                    <h2 className='font-bold text-3xl '>Headphones</h2>
                    <span>Electronics</span>
                </div>
                <span>Price: $60</span>
                <div>
                    <span className='font-bold text-xl'>Description:</span>
                    <p className='text-lg'>Lorem ipsum dolor sit, amet consectetur adipisicing elit.
                        Voluptatem facilis cupiditate reiciendis dignissimos hic aliquam esse aspernatur quam similique libero!</p>
                </div>
                <CustomButton paddingX='px-4' paddingY='py-4'>Email Owner</CustomButton>
            </div>
        </div>
    )
}

export default ItemDetailsPage