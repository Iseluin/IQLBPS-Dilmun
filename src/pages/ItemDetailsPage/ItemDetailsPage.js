import React from 'react'
import CustomButton from '../../components/CustomComponents/CustomButton'

const ItemDetailsPage = () => {
    return (
        <div className='md:flex md:h-screen text-background p-6 bg-background mt-11'>
            <div>
                <img className='max-h-full mb-6 md:mb-0' src="https://images.unsplash.com/photo-1553062407-98eeb64c6a62?ixlib=rb-4.0.3&ixid=MnwxMjA3fDB8MHxzZWFyY2h8NHx8YmFnfGVufDB8fDB8fA%3D%3D&w=1000&q=80" alt="headphones" />
            </div>
            <div className=' md:max-w-lg md:ml-auto min-h-[600px] md:mr-auto flex flex-col gap-7 rounded pt-5 px-5 bg-primary '>
                <div>
                    <h2 className='font-bold text-3xl '>Backpack</h2>
                    <span>Other</span>
                </div>
                <span className='flex gap-1'><p className='font-semibold'>Price:</p> $60</span>
                <span className='flex gap-1'><p className='font-semibold'>Item Type:</p> New</span>
                <span className='flex gap-1'><p className='font-semibold'>Country:</p> Spain</span>
                <span className='flex gap-1'><p className='font-semibold'>Address:</p> C/Conde de Xiquena 7, 3ยบ, 1A Izda, 280280, Madrid</span>
                <div>
                    <span className='font-bold text-xl'>Description:</span>
                    <p className='text-lg'>Lorem ipsum dolor sit, amet consectetur adipisicing elit.
                        Voluptatem facilis cupiditate reiciendis dignissimos hic aliquam esse aspernatur quam similique libero!</p>
                </div>
                <CustomButton paddingX='px-4' paddingY='py-4'>Email Owner</CustomButton>
            </div>
        </div>
    )
}

export default ItemDetailsPage