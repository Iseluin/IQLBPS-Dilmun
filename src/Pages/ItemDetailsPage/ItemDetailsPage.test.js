import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import ItemDetailsPage from './ItemDetailsPage';
import { store } from '../../app/store';

test('Sort the Products', () => {
    const tree = render(<Provider store={store}><ItemDetailsPage /></Provider>);
    expect(tree).toMatchSnapshot();
});
