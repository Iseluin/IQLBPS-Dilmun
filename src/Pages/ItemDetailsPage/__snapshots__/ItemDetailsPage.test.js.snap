// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sort the Products 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="md:flex h-full text-background p-6 bg-background gap-4 mt-11"
        data-testid="item-details-page"
      >
        <div>
          <img
            alt="headphones"
            class="max-h-full min-h-[600px] mb-6 md:mb-0"
            src="https://images.unsplash.com/photo-1553062407-98eeb64c6a62?ixlib=rb-4.0.3&ixid=MnwxMjA3fDB8MHxzZWFyY2h8NHx8YmFnfGVufDB8fDB8fA%3D%3D&w=1000&q=80"
          />
        </div>
        <div
          class=" md:max-w-lg md:ml-auto min-h-[700px] max-h-full md:mr-auto flex flex-col gap-7 rounded pt-5 px-5 bg-primary "
        >
          <div>
            <h2
              class="font-bold text-3xl "
            >
              Backpack
            </h2>
            <span>
              Other
            </span>
          </div>
          <span
            class="flex gap-1"
          >
            <p
              class="font-semibold"
            >
              items_details.price
            </p>
             $60
          </span>
          <span
            class="flex gap-1"
          >
            <p
              class="font-semibold"
            >
              items_details.item_type
            </p>
             New
          </span>
          <span
            class="flex gap-1"
          >
            <p
              class="font-semibold"
            >
              items_details.country
            </p>
             Spain
          </span>
          <span
            class="flex gap-1"
          >
            <p
              class="font-semibold"
            >
              items_details.address
            </p>
             C/Conde de Xiquena 7, 3ยบ, 1A Izda, 280280, Madrid
          </span>
          <div>
            <span
              class="font-bold text-xl"
            >
              items_details.description
            </span>
            <p
              class="text-lg"
            >
              Lorem ipsum dolor sit, amet consectetur adipisicing elit. Voluptatem facilis cupiditate reiciendis dignissimos hic aliquam esse aspernatur quam similique libero!
            </p>
          </div>
          <button
            class="bg-tertiary px-4 py-4 w-full lg:w-auto text-white font-semibold rounded-lg hover:shadow-lg hover:bg-secondary transition duration-500 cursor-pointer"
            type="submit"
          >
            items_details.email_owner
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="md:flex h-full text-background p-6 bg-background gap-4 mt-11"
      data-testid="item-details-page"
    >
      <div>
        <img
          alt="headphones"
          class="max-h-full min-h-[600px] mb-6 md:mb-0"
          src="https://images.unsplash.com/photo-1553062407-98eeb64c6a62?ixlib=rb-4.0.3&ixid=MnwxMjA3fDB8MHxzZWFyY2h8NHx8YmFnfGVufDB8fDB8fA%3D%3D&w=1000&q=80"
        />
      </div>
      <div
        class=" md:max-w-lg md:ml-auto min-h-[700px] max-h-full md:mr-auto flex flex-col gap-7 rounded pt-5 px-5 bg-primary "
      >
        <div>
          <h2
            class="font-bold text-3xl "
          >
            Backpack
          </h2>
          <span>
            Other
          </span>
        </div>
        <span
          class="flex gap-1"
        >
          <p
            class="font-semibold"
          >
            items_details.price
          </p>
           $60
        </span>
        <span
          class="flex gap-1"
        >
          <p
            class="font-semibold"
          >
            items_details.item_type
          </p>
           New
        </span>
        <span
          class="flex gap-1"
        >
          <p
            class="font-semibold"
          >
            items_details.country
          </p>
           Spain
        </span>
        <span
          class="flex gap-1"
        >
          <p
            class="font-semibold"
          >
            items_details.address
          </p>
           C/Conde de Xiquena 7, 3ยบ, 1A Izda, 280280, Madrid
        </span>
        <div>
          <span
            class="font-bold text-xl"
          >
            items_details.description
          </span>
          <p
            class="text-lg"
          >
            Lorem ipsum dolor sit, amet consectetur adipisicing elit. Voluptatem facilis cupiditate reiciendis dignissimos hic aliquam esse aspernatur quam similique libero!
          </p>
        </div>
        <button
          class="bg-tertiary px-4 py-4 w-full lg:w-auto text-white font-semibold rounded-lg hover:shadow-lg hover:bg-secondary transition duration-500 cursor-pointer"
          type="submit"
        >
          items_details.email_owner
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
